tests/framework/cli/test_starters.py

```diff
@@ -78,10 +82,46 @@ def test_starter_list(fake_kedro_cli):
     result = CliRunner().invoke(fake_kedro_cli, ["starter", "list"])
 
     assert result.exit_code == 0, result.output
-    for alias in _STARTER_ALIASES:
+    for alias in _OFFICIAL_STARTER_SPECS:
         assert alias in result.output
 
 
+def test_starter_list_with_starter_plugin(fake_kedro_cli, entry_point):
+    """Check that `kedro starter list` prints out the plugin starters."""
+    entry_point.load.return_value = [KedroStarterSpec("valid_starter", "valid_path")]
+    entry_point.module = "valid_starter"
+    result = CliRunner().invoke(fake_kedro_cli, ["starter", "list"])
+    assert result.exit_code == 0, result.output
+    assert "valid_starter" in result.output
```

Comment:
Minor suggestion: this test could be made a bit stricter since `assert "valid_starter" in result.output` is quite weak, especially when "valid_starter" is both the starter alias and the module name.

So maybe set `entry_point.module = "valid_starter_module"` and test the `result.output` a bit more thoroughly.